trigger/src/main/scala/core/api/rest/SpringBootConfig.scala

```diff
@@ -0,0 +1,65 @@
+package core.api.rest
+
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind._
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import core.models.enums.{EventTypes, JobStatuses, JobTypes}
+import core.models.enums.EventTypes.EventType
+import core.models.enums.JobStatuses.JobStatus
+import core.models.enums.JobTypes.JobType
+import core.persistance.{EventRepositoryImpl, WorkflowRepositoryImpl}
+import core.service.{EventsService, EventsServiceImpl, WorkflowService, WorkflowServiceImpl}
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.context.annotation.Bean
+
+@SpringBootApplication
+class SpringBootConfig {
+
+  @Bean
+  def objectMapper(): ObjectMapper = {
+    val module = new SimpleModule()
+      .addDeserializer(classOf[EventType], new EventTypesDeserializer)
+      .addDeserializer(classOf[JobStatus], new JobStatusesDeserializer)
+      .addDeserializer(classOf[JobType], new JobTypesDeserializer)
+
+    new ObjectMapper()
+      .registerModule(DefaultScalaModule)
+      .registerModule(new JavaTimeModule())
+      .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+      .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+      .configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, true)
+      .registerModule(module)
+  }
+
+  @Bean
+  def getEventService: EventsService = new EventsServiceImpl(new EventRepositoryImpl)
+  @Bean
+  def getWorkflowService: WorkflowService = new WorkflowServiceImpl(new WorkflowRepositoryImpl)
+
+}
+
+class EventTypesDeserializer extends JsonDeserializer[EventType] {
+  override def deserialize(p: JsonParser, ctxt: DeserializationContext): EventType = {
+    val node = p.getCodec.readTree[JsonNode](p)
+    val value = node.get("name").textValue()
+    EventTypes.eventTypes.find(_.name == value).getOrElse(throw new Exception("Failed to find enum value"))
+  }
+}
+
+class JobStatusesDeserializer extends JsonDeserializer[JobStatus] {
+  override def deserialize(p: JsonParser, ctxt: DeserializationContext): JobStatus = {
+    val node = p.getCodec.readTree[JsonNode](p)
+    val value = node.get("name").textValue()
```

Comment:
Maybe extract the literal into a companion object or something similar?

Reply:
I am not sure about this one. I cant find good place where to put literal.